common/cluster/archival_status.go

```diff
@@ -47,38 +48,32 @@ const (
 )
 
 // NewArchivalConfig constructs a new valid ArchivalConfig
-func NewArchivalConfig(status ArchivalStatus, defaultBucket string, enableReadFromArchival bool) *ArchivalConfig {
+func NewArchivalConfig(
+	status ArchivalStatus,
+	enableReadFromArchival bool,
+	domainDefaultStatus ArchivalStatus,
+	domainDefaultURI string,
+) *ArchivalConfig {
 	ac := &ArchivalConfig{
 		status:                 status,
-		defaultBucket:          defaultBucket,
 		enableReadFromArchival: enableReadFromArchival,
+		domainDefaultStatus:    domainDefaultStatus,
+		domainDefaultURI:       domainDefaultURI,
 	}
-	if !ac.isValid() {
-		return &ArchivalConfig{
-			status:                 ArchivalDisabled,
-			defaultBucket:          "",
-			enableReadFromArchival: false,
-		}
+	if !ac.isDomainDefaultValid() {
+		ac.domainDefaultStatus = ArchivalDisabled
```

Comment:
If the domain is not valid lets also set the configs for cluster: status and enableReadFromArchival to their disabled states. To make this more clear lets include a private helper method `disabledConfig()`